ibm/resource_ibm_dl_gateway_virtual_connection.go

```diff
@@ -0,0 +1,279 @@
+			dlGatewayId: {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The Direct Link gateway identifier",
+			},
+			dlVCType: {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: InvokeValidator("ibm_dl_gateway_vc", dlVCType),
+				Description:  "The user-defined name for this virtual connection. Virtualconnection names are unique within a gateway. This is the name of thevirtual connection itself, the network being connected may have its ownname attribute",
+			},
+			dlVCName: {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     false,
+				ValidateFunc: validateRegexpLen(1, 63, "^([a-zA-Z]|[a-zA-Z][-_a-zA-Z0-9]*[a-zA-Z0-9])$"),
+				Description:  "The type of virtual connection.Allowable values",
+			},
+			dlVCNetworkId: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: "Unique identifier of the target network. For type=vpc virtual connections this is the CRN of the target VPC. This field does not apply to type=classic connections.",
+			},
+			dlVCCreatedAt: {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			dlVCStatus: {
+				Type:     schema.TypeString,
+				Computed: true,
+				ForceNew: false,
+			},
+
+			dlVCNetworkAccount: {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+func resourceIBMdlGatewayVCValidator() *ResourceValidator {
+
+	validateSchema := make([]ValidateSchema, 1)
+	vcType := "classic, vpc"
+	validateSchema = append(validateSchema,
+		ValidateSchema{
+			Identifier:                 dlVCType,
+			ValidateFunctionIdentifier: ValidateAllowedStringValue,
+			Type:                       TypeString,
+			Required:                   true,
+			AllowedValues:              vcType})
+	ibmDLGatewayVCResourceValidator := ResourceValidator{ResourceName: "ibm_dl_gateway_vc", Schema: validateSchema}
+	return &ibmDLGatewayVCResourceValidator
+}
+func resourceIBMdlGatewayVCCreate(d *schema.ResourceData, meta interface{}) error {
+	directLink, err := directlinkClient(meta)
+	if err != nil {
+		return err
+	}
+
+	createGatewayVCOptions := &directlinkapisv1.CreateGatewayVirtualConnectionOptions{}
+
+	gatewayId := d.Get(dlGatewayId).(string)
+	createGatewayVCOptions.SetGatewayID(gatewayId)
+	vcName := d.Get(dlVCName).(string)
+	createGatewayVCOptions.SetName(vcName)
+	vcType := d.Get(dlVCType).(string)
+	createGatewayVCOptions.SetType(vcType)
+
+	if _, ok := d.GetOk(dlVCNetworkId); ok {
+		vcNetworkId := d.Get(dlVCNetworkId).(string)
+		createGatewayVCOptions.SetNetworkID(vcNetworkId)
+	}
+
+	gatewayVC, response, err := directLink.CreateGatewayVirtualConnection(createGatewayVCOptions)
+	if err != nil {
+		log.Printf("[DEBUG] Create Direct Link Gateway (Dedicated) Virtual connection err %s\n%s", err, response)
+		return err
+	}
+	d.SetId(*gatewayVC.ID)
```

Comment:
Set the ID as combination of gatewayID/virtualconnectionID

Add new schema attribute virtual_connection_id and set the virtualconenction ID

d.SetId(fmt.Sprintf("%s/%s", gatewayId, *gatewayVC.ID))
d.SetId("virtual_connection_id", *gatewayVC.I)